internal/provider/data_source_users.go

```diff
@@ -0,0 +1,156 @@
+package provider
+
+import (
+	"context"
+	"fmt"
+	"strconv"
+	"time"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+
+	"github.com/infrahq/infra/api"
+)
+
+func dataSourceUsers() *schema.Resource {
+	return &schema.Resource{
+		Description: "Get a list of registered Infra users.",
+
+		ReadContext: dataSourceUsersRead,
+
+		Schema: map[string]*schema.Schema{
+			"id": &schema.Schema{
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"filter": &schema.Schema{
+				Type:     schema.TypeList,
+				Optional: true,
+				MaxItems: 1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"name": &schema.Schema{
+							Description:      "The name of the user to find.",
```

Comment:
```suggestion
							Description:      "The name of the user.",
```

Small nit but should these end in `.`?